Assets/Project/Scripts/Utils/SoundManager.cs

```diff
@@ -0,0 +1,271 @@
+using Project.Scripts.Utils.Patterns;
+using Project.Scripts.Utils.PlayerPrefsUtils;
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+using UnityEngine;
+
+namespace Project.Scripts.Utils
+{
+    public enum EBGMKey {
+        BGM_Gameplay
+    }
+
+    public enum ESEKey {
+        SE_Success,
+        SE_Failure,
+        SE_ThunderAttack,
+    }
+
+    public class SoundManager : SingletonObject<SoundManager>
+    {
+        /// <summary>
+        /// 同時再生できるSEの数
+        /// </summary>
+        private const int _MAX_SE_NUM = 8;
+
+        /// <summary>
+        /// 複数SE同時再生を考慮し、SE用のAudioSourceを複数用意する
+        /// </summary>
+        private AudioSource[] _SePlayers;
+
+        /// <summary>
+        /// BGM再生用のAudioSource
+        /// </summary>
+        private AudioSource _BgmPlayer;
+
+        /// <summary>
+        /// BGMリスト
+        /// </summary>
+        [SerializeField] private List<SoundData> _bgmClipList;
+
+        /// <summary>
+        /// SEリスト
+        /// </summary>
+        [SerializeField] private List<SoundData> _seClipList;
+
+        /// <summary>
+        /// 初期BGMボリューム
+        /// </summary>
+        [SerializeField] private float _INITIAL_BGM_VOLUME = 0.25f;
+
+        /// <summary>
+        /// 初期SEボリューム
+        /// </summary>
+        [SerializeField] private float _INITIAL_SE_VOLUME = 1.0f;
+
+        private void Awake()
+        {
+            // BGM再生用のAudioSourceをアタッチする
+            _BgmPlayer = gameObject.AddComponent(typeof(AudioSource)) as AudioSource;
+            _BgmPlayer.loop = true;
+
+            // SE再生用のAudioSourceをアタッチする
+            _SePlayers = new AudioSource[_MAX_SE_NUM];
+            for (var i = 0; i < _MAX_SE_NUM; i++) {
+                var player = gameObject.AddComponent(typeof(AudioSource)) as AudioSource;
+                player.loop = false;
+
+                _SePlayers[i] = player;
+            }
+
+            DontDestroyOnLoad(gameObject);
+        }
+
+        /// <summary>
+        /// SEを再生する
+        /// </summary>
+        /// <param name="key"> 再生するSEのキー </param>
+        public void PlaySE(ESEKey key)
+        {
+            var clip = GetSEClip(key);
+            if (clip == null)
+                return;
+
+            if (_SePlayers.Any(src => !src.isPlaying)) {
+                // 再生していないAudioSourceを探す
+                var player = _SePlayers.First(source => !source.isPlaying);
+
+                player.PlayOneShot(clip);
+            } else {
+                Debug.LogWarning($"Failed to Play SE: {key} because there is no available audio source");
+            }
+        }
+
+        /// <summary>
+        /// SEを停止する
+        /// </summary>
+        /// <param name="key"> 停止するSEのキー </param>
+        public void StopSE(ESEKey key)
+        {
+            var clip = GetSEClip(key);
+            if (clip == null)
+                return;
+
+            var player = _SePlayers.Where(src => src.clip != null).SingleOrDefault(src => src.clip.name == clip.name);
```

Comment:
```suggestion
            var player = _SePlayers
                .Where(src => src.clip != null)
                .SingleOrDefault(src => src.clip.name == clip.name);
```

今更だけど，こう書く，もしくは

```suggestion
            var player = _SePlayers.Where(src => src.clip != null)
                .SingleOrDefault(src => src.clip.name == clip.name);
```

こう書く方がみやすい

Reply:
他同様